components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/ContextMenuCandidate.kt

```diff
@@ -106,6 +107,27 @@ data class ContextMenuCandidate(
             }
         )
 
+        /**
+         * Context Menu item: "Open Link in external App".
+         */
+        fun createOpenInExternalAppCandidate(
+            context: Context,
+            appLinksUseCases: AppLinksUseCases
+        ) = ContextMenuCandidate(
+                id = "mozac.feature.contextmenu.open_in_external_app",
+                label = context.getString(R.string.mozac_feature_contextmenu_open_link_in_external_app),
+                showFor = { _, hitResult -> hitResult.canOpenInExternalApp(appLinksUseCases) },
+                action = { _, hitResult ->
+                    val link = hitResult.getLink()
+                    val redirect = appLinksUseCases.appLinkRedirect(link)
+                    val intent = redirect.appIntent
```

Comment:
The issue mentioned market intent.  Should we check if `appIntent` is null, if there is a `marketplaceIntent`?

Reply:
Added :)

Reply:
I have tried to make it work so that the marketplace is opened with the correct app, when it is not installed, but I could never get `createBrowsableIntents` to return a `marketPlaceIntent` for the https links I've tried somehow.

Reply:
I used one of the links in this page https://paul.kinlan.me/sharing-natively-on-android-from-the-web/. There is a intent link in this page that targets `com.chrome.beta` package.  Try this out and see if you can redirect to Google playstore.

Reply:
Ah you mean these links. Yeah I've already tried them and get my context menu entry working for them. But I think there's more problems with these kind of links because no other context menu entries will be shown for them (see screenshot). And also when you enable "Open links in apps" from the menu and click on one of them the current master crashes for me. Do you have the same problem?

![grafik](https://user-images.githubusercontent.com/9115235/72647682-636aef80-3979-11ea-91ac-50de6fc5ee1b.png)

Reply:
Interesting, I'm seeing this as well.  I will open an issue for this, thanks for pointing it out.  I am only seeing this issue in sample browser.   Are you seeing it on sample browser as well?

Reply:
Yep. I'm seeing this in sample browser.

Reply:
Issue opened.  https://github.com/mozilla-mobile/android-components/issues/5637. Thanks!

Reply:
I've put up a fix.  Can you check with this fix you can get marketplace intents to work with context menu as well?  Thanks,

Reply:
> Ah you mean these links. Yeah I've already tried them and get my context menu entry working for them. But I think there's more problems with these kind of links because no other context menu entries will be shown for them (see screenshot). 

I think its fine if there is only one context menu entry in this case.  The experience will be more consistent if the context menu always pops up on long presses.

Reply:
I merged your branch into this one and made the changes. 👍